plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/internal/CloudSdkContextFunction.java

```diff
@@ -61,11 +61,9 @@ public Object compute(IEclipseContext context, String contextKey) {
 
     Object path = context.get(PreferenceConstants.CLOUDSDK_PATH);
     File location = toFile(path);
```

Comment:
CloudSdkProvider now takes an additional location that can come from the context. If there is no context location, it uses the one in the preference store. If none there either, it then delegates SDK discovery to the library. If the library can't find it, then null is returned and later on the user is prompted to set a valid one.